### PR TITLE
[K4.0] Backend: Edit less

### PR DESCRIPTION
**Pull request by 810**
Todo
- [x] Fix model, filename isn't working both css and less
- [x] create custom.less
- [ ] Fix select template, now you must click first edit, then go back and you can edit css/less

### CODE REVIEW

**Comment by rich20** [comment]
If you want edit .less or .css. 

Warning: Invalid argument supplied for foreach() in components/com_kunena/template/joomla30/templates/chooseless.php on line 43
/components/com_kunena/template//less 


Warning: Invalid argument supplied for foreach() in components/com_kunena/template/joomla30/templates/choosecss.php on line 44
/components/com_kunena/template//css

**Comment by rich20** [comment]
#### Joomla 2.5
-  [x] These options are not available

#### Joomla 3.4
##### edit .css
Warning JFolder: :files: Path is not a folder. Path: /components/com_kunena/template/css

Warning: fopen(/components/com_kunena/template//css/custom.css): failed to open stream: No such file or directory in components/com_kunena/views/templates/view.html.php on line 102

Warning: fwrite() expects parameter 1 to be resource, boolean given in components/com_kunena/views/templates/view.html.php on line 103

Warning: fclose() expects parameter 1 to be resource, boolean given in components/com_kunena/views/templates/view.html.php on line 104

Warning: Invalid argument supplied for foreach() in components/com_kunena/template/joomla30/templates/choosecss.php on line 44

/components/com_kunena/template//css Permissions Of CSS Files
##### edit .less
Warning JFolder: :files: Path is not a folder. Path: /components/com_kunena/template/less

Warning: fopen(/components/com_kunena/template//less/custom.less): failed to open stream: No such file or directory in components/com_kunena/views/templates/view.html.php on line 68

Warning: fwrite() expects parameter 1 to be resource, boolean given in components/com_kunena/views/templates/view.html.php on line 69

Warning: fclose() expects parameter 1 to be resource, boolean given in components/com_kunena/views/templates/view.html.php on line 70

Warning: Invalid argument supplied for foreach() in components/com_kunena/template/joomla30/templates/chooseless.php on line 43

/components/com_kunena/template//less Permissions Of LESS Files

File: components/com_kunena/admin/controllers/templates.php [hunk 1]
```diff
@@ -252,18 +252,156 @@ function uninstall()
 		$this->setRedirect(KunenaRoute::_($this->baseurl, false));
 	}
 
+	function chooseless()
+	{
+		$jinput = JFactory::getApplication()->input;
+		$template = $jinput->get('id', '', 'method', 'cmd');
```
Comment (xillibit) [inline components/com_kunena/admin/controllers/templates.php]: You can just do here $this->app->input->get('id', '', 'method', 'cmd'); you don't need to call 
$jinput = JFactory::getApplication()->input;

**Comment by xillibit** [comment]
There is a string missing : 

ddre

The right of file has changed after to have saved it : 

dssdqdsqse

**Comment by 810** [comment]
code updated

**Comment by xillibit** [comment]
Now i have the same issue taht rich : https://github.com/Kunena/Kunena-Forum/pull/3060#issuecomment-100020483

**Comment by 810** [comment]
#### Edit Less
- [x] Load right template
- [x] create custom.less
- [x] save/apply working
- [ ] Hide less for none hmvc template

#### Edit Css
- [x] Load right template
- [x] create custom.css
- [x] load custom.css
- [x] save/apply working

**Comment by xillibit** [comment]
> Hide less for none hmvc template

Where hide LESS ?

**Comment by 810** [comment]
Show msg on chooseless, 

so select blue eagle, click on edit less.

Now the custom.less is created on the blue eagle template

**Comment by xillibit** [comment]
The custom.less is created before you click on edit less or when you have clicked on edit less ?

You have a method in template to check if it's HMVC: isHmvc()

**Comment by 810** [comment]
When you go to the chooseless. So edit less button

**Comment by rich20** [comment]
I've several error messages with selected Blue Eagle and edit less. Maybe it would be better the buttons would be only available after click on edit on your chosen template.
bildschirmfoto vom 2015-05-13 03 24 25

**Comment by rich20** [comment]
Next issue : You can't empty a edited custom.file (CSS or LESS). Click on save closes the windows. Minimum is a space then you can save it.

**Comment by 810** [comment]
fixed the issues

**Comment by rich20** [comment]
- [x] Empy a edited custom file - Success
- [x] Try to edit a less.file in Blue Eagle - Success
- [ ] It shows only the code COM_KUNENA_A_TEMPLATE_MANAGER_NO_LESS

**Comment by 810** [comment]
yeah saw that too,  but I don't know why it isn't loading the string.

**Pull request merged**